src/pretix/plugins/sendmail/models.py

```diff
@@ -280,9 +281,10 @@ def save(self, **kwargs):
         if self.event.has_subevents:
             for se in self.event.subevents.annotate(has_sm=Exists(ScheduledMail.objects.filter(
                     subevent=OuterRef('pk'), rule=self))).filter(has_sm=False):
-                sm = ScheduledMail(rule=self, subevent=se, event=self.event)
-                sm.recompute()
-                create_sms.append(sm)
+                if not self.subevent or self.subevent == se:
```

Comment:
1. This **is** a correct implementation, but a little inefficient – there's probably no reason to have a `for` loop over **all** subevents when we already know we only need one?
2. If ``self.subevent`` **changes**, we also need to clean up ScheduledMail instances for other subevents that have previously existed.